pkg/overlay/overlay_test.go

```diff
@@ -62,3 +63,28 @@ func TestOverlayLookup(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, r)
 }
+
+func TestOverlayBulkLookup(t *testing.T) {
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", 0))
```

Comment:
Start test-servers on `127.0.0.1:0`, other ports may be blocked or other hosts may trigger firewall.